src/BenchmarkDotNet.Core/Diagnosers/MemoryDiagnoser.cs

```diff
@@ -62,13 +62,28 @@ public bool IsAvailable(Summary summary)
             public bool AlwaysShow => true;
             public ColumnCategory Category => ColumnCategory.Diagnoser;
             public int PriorityInCategory => 0;
+            public QuantityType QuantityType => QuantityType.Size;
+            public string GetValue(Summary summary, Benchmark benchmark) => GetValue(summary, benchmark, SummaryStyle.Default);
 
-            public string GetValue(Summary summary, Benchmark benchmark)
+            public string GetValue(Summary summary, Benchmark benchmark, ISummaryStyle style)
             {
                 if (!results.ContainsKey(benchmark) || benchmark.Job.Env.Runtime is MonoRuntime)
                     return "N/A";
 
-                return results[benchmark].BytesAllocatedPerOperation.ToFormattedBytes();
+                var value = results[benchmark].BytesAllocatedPerOperation;
+                return QuantityType == QuantityType.Size ? value.ToSizeStr(style.SizeUnit, 1, style.PrintUnitsInContent) : ((double)value).ToStr();
```

Comment:
I thought exactly the same thing, but then I realized that this method is used for display only, and previously we also used `double`. Specifically, `ToFormattedBytes` did
 `double dblSByte = bytes;`
` dblSByte = bytes / (double)BytesInKiloByte;`
`return String.Format...dblSByte`